Add new licenses and license updates

AyanSinhaMahapatra opened the pull request:
References: 
* https://github.com/aboutcode-org/scancode-toolkit/issues/3938
* https://github.com/aboutcode-org/scancode-toolkit/issues/3949
* https://github.com/aboutcode-org/scancode-toolkit/issues/3973
* https://github.com/aboutcode-org/scancode-toolkit/issues/3990
* https://github.com/aboutcode-org/scancode-toolkit/issues/3991
* https://github.com/aboutcode-org/scancode-toolkit/issues/3992
* https://github.com/aboutcode-org/scancode-toolkit/issues/3993


### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁
* [x] Updated documentation pages (if applicable)
* [x] Updated CHANGELOG.rst (if applicable)

AyanSinhaMahapatra committed:
Add new licenses and updates

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra commented on src/licensedcode/data/licenses/freetos-mit.LICENSE:
```diff
@@ -0,0 +1,32 @@
+---
+key: freetos-mit
```
@DennisClark maybe a typo here?
```suggestion
key: freertos-mit
```

DennisClark replied:
oops

DennisClark replied:
I will fix that in DejaCode shortly

DennisClark replied:
DejaCode updates: 
* deprecated freetos-mit, setting it to inactive (by design, cannot change the license key in DejaCode)
* created freertos-mit

AyanSinhaMahapatra committed:
Add fixes to license metadata

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add new licenses added to dejacode

Added by @DennisClark, thanks to @armijnhemel and @pombredanne
for reporting these too.

Reference: https://github.com/aboutcode-org/scancode-toolkit/issues/3992
Reference: https://github.com/aboutcode-org/scancode-toolkit/issues/3991
Reference: https://github.com/aboutcode-org/scancode-toolkit/issues/3990
Reference: https://github.com/aboutcode-org/scancode-toolkit/issues/3993
Reference: https://github.com/aboutcode-org/scancode-toolkit/issues/3973
Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

pombredanne reviewed (approved):
Looking good. Merging!
